### PR TITLE
chore: Remove go server references

### PR DESCRIPTION
**Pull request by achals**
**What this PR does / why we need it**:

As title. This cleans up dead/unused code from Feast.

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: sdk/python/feast/feature_store.py [hunk 1]
```diff
@@ -2283,45 +2226,12 @@ def serve(
...
-                    port,
-                    enable_logging=enable_logging,
-                    logging_options=logging_options,
-                )
-            else:
-                raise ValueError(
-                    f"Unsupported server type '{type_}'. Must be one of 'http' or 'grpc'."
-                )
-        else:
-            if type_ != "http":
```
Comment (kevjumba) [inline sdk/python/feast/feature_store.py:2319]: this probably still needs to exizt?

**Review by kevjumba — APPROVED** [review]

File: sdk/python/feast/feature_store.py (hunk 1 above)
Comment (achals) [reply sdk/python/feast/feature_store.py:2319]: It's there actually - look up at line 2229 in the new version.

**Comment by judahrand** [comment]
Any reason to not also remove the `embedded_go` module?

**Pull request merged**